server/etcdserver/api/membership/storev2.go

```diff
@@ -78,6 +78,13 @@ func TrimMembershipFromV2Store(lg *zap.Logger, s v2store.Store) error {
 	return nil
 }
 
+func verifyNoMembersInStore(lg *zap.Logger, s v2store.Store) {
+	members, removed := membersFromStore(lg, s)
+	if len(members) != 0 || len(removed) != 0 {
+		lg.Panic("store has membership info")
+	}
+}
```

Comment:
This verification makes sense, but should be in a following PR which removes all the code of adding/removing member in v2store.

Reply:
@ahrtr this is being called only from the RaftCluster Store function which must get an empty store and is only invoked from GetMembershipInfoInV2Format today. This verification is added because the code to assume that member deletion from the store is not needed - it directly adds the removed members to the RemovedMembers prefix in the store. 
I can move the verification to subsequent PR as per review feedback but wanted to clarify that it is protecting the correctness of logic to handle removed members. Please let me know if you would still like me to move this out.

Reply:
> @ahrtr this is being called only from the RaftCluster Store function which must get an empty store

Yes, I agree. But the `(*RaftCluster) Store` will be called in multiple places in following PR(s). The verification will definitely fail **after** you add more calling of the Store and **before** you remove the code of adding/removing members data in v2store. Currently you only call the `Store` in the etcdutl backup command, so you do not see any verification failure.

A better way should be get it included in a follow PR as the first commit:
- add the verification. We should get a failed verification;
- remove all the code of adding/removing/updating in v2store in the applying code path. Then the verification should be all good.

Since we will resolve the big task "v2store deprecation" in lots of small PRs, let's try to get each PR as simple and standalone as possible.

Reply:
>>add the verification. We should get a failed verification;
Adding more clarification - The Store method will not get called directly. It will get called from GetMembershipInfoInV2Format  which should pass empty store as an argument to Store method. The verification should not fail.

Reply:
ack. I messed this v2store with [server.go#L252](https://github.com/etcd-io/etcd/blob/9f72c648d3829810f4256ed1868fcf1c7665aa44/server/etcdserver/server.go#L252)